catalog/src/main/kotlin/com/adevinta/spark/catalog/datastore/buttonconfigurator/ButtonsConfiguratorPropertiesSerializer.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2023 Adevinta
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.adevinta.spark.catalog.datastore.buttonconfigurator
+
+import androidx.datastore.core.Serializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.Json
+import java.io.InputStream
+import java.io.OutputStream
+
+internal object ButtonsConfiguratorPropertiesSerializer : Serializer<ButtonsConfiguratorProperties> {
```

Comment:
I think you don't need this class.
You can directly use something like that

```kotlin
        private val Context.datastore: DataStore<ButtonsConfiguratorProperties> by dataStore(
            fileName = "buttons_configurator_properties.pb",
            serializer = ButtonsConfiguratorProperties.serializer(),
        )
```

Reply:
Oh, sorry, it's a `androidx.datastore.core.Serializer` not a KotlinX Serializer.